python/pyiceberg/io/pyarrow.py

```diff
@@ -35,13 +36,14 @@
     OutputFile,
     OutputStream,
 )
+from pyiceberg.typedef import Properties
 
 
 class PyArrowFile(InputFile, OutputFile):
     """A combined InputFile and OutputFile implementation that uses a pyarrow filesystem to generate pyarrow.lib.NativeFile instances
 
     Args:
-        location(str): A URI or a path to a local file
```

Comment:
I now agree with you on this point and will change back to "location"

originally I changed the name because the value being passed here is explicitly *not* a URI anymore (per the from_uri() call in pyarrow) so I thought a different name would be better. But 'location' already means uri OR path.